dotnet/src/SemanticKernel.Core/Functions/KernelFunctionFromPrompt.cs

```diff
@@ -66,7 +66,7 @@ internal sealed class KernelFunctionFromPrompt : KernelFunction
 
         if (executionSettings is not null)
         {
-            promptConfig.ExecutionSettings.Add(executionSettings);
+            promptConfig.ExecutionSettings.Add(executionSettings.ServiceId ?? PromptExecutionSettings.DefaultServiceId, executionSettings);
```

Comment:
There are a couple of places with this logic `executionSettings.ServiceId ?? PromptExecutionSettings.DefaultServiceId` and there is a chance to miss it somewhere. Should we convert `ServiceId` to non-nullable and set it to `default` on `PromptExecutionSettings` creation? In this case we will be sure that `ServiceId` is always configured, and it will be possible for user to set different value if needed.

```csharp
public class PromptExecutionSettings
{
    public const string DefaultServiceId = "default";

    [JsonPropertyName("service_id")]
    public string ServiceId { get; set; } = DefaultServiceId;

    // Other properties
```

Reply:
`ServiceId` will be eliminated and doing that now results in a bunch of unit test changes that will need to be reverted.